lib/usd/ui/layerEditor/layerTreeModel.cpp

```diff
@@ -151,7 +151,7 @@ bool LayerTreeModel::canDropMimeData(
     }
 
     auto parentItem = layerItemFromIndex(parentIndex);
-    if (!parentItem || parentItem->isReadOnly()) {
+    if (!parentItem || parentItem->isReadOnly() || parentItem->isLocked()) {
```

Comment:
By design, immediate children of a locked parent layer cannot be reorganised (also covers the system lock layers by the shared constraint of PermissionToEdit = false).